### ISSUE TITLE
Add a "Add Note" form to all `/db/` `show.erb` views

### ISSUE THREAD

**Issue by postmodern** [ronin-rb/ronin-app#43]
Add a "Add Note" text area and form to all `/db/``show.erb`  views to allow adding a new comment to the `notes` association of the record. This may also require adding a `_add_note.erb` partial template.

**Comment by AI-Mozi** [comment]
Ok, its not implemented yet.
Is there a reason why `Note` doesn't belongs_to  all `/db` models ?

**Comment by postmodern** [comment]
@AI-Mozi `Note` is supposed to `belongs_to` any non-join-models that the user would interact with, such as `IPAddress`, `HostName`, `Service`, `URL`, maybe even `OpenPort`. I am in the process of adding more new models in the `osint` branch of `ronin-db-activerecord`, and will have to rebase the branch and update `Note` to have more `has_many` associations to the new models.

**Comment by postmodern** [comment]
It occurred to me you could DRY up the `/db/.../notes` routes by dynamically defining them:

```ruby
{ip_address: Ronin::DB::IPAddress, ...}.each do |name,model|
  post "/db/#{name}/:id/notes" do
    @record = model.find(params[:id])
    ...
  end

  delete "/db/#{name}/:id/notes/:note_id" do
    @record = model.find(params[:id])
    
    if @record
      if @record.notes.destroy(params[:note_id])
        ...
       else
        ...
      end
    end
  end
end
```

**Comment by postmodern** [comment]
Implemented by PR #59.
